cpp/src/parquet/column_reader.cc

```diff
@@ -329,17 +329,16 @@ void SerializedPageReader::InitDecryption() {
 }
 
 void SerializedPageReader::UpdateDecryption(const std::shared_ptr<Decryptor>& decryptor,
-                                            int8_t module_type,
-                                            const std::string& page_aad) {
+                                            int8_t module_type, std::string* page_aad) {
   DCHECK(decryptor != nullptr);
   if (crypto_ctx_.start_decrypt_with_dictionary_page) {
     std::string aad = encryption::CreateModuleAad(
         decryptor->file_aad(), module_type, crypto_ctx_.row_group_ordinal,
         crypto_ctx_.column_ordinal, kNonPageOrdinal);
     decryptor->UpdateAad(aad);
   } else {
-    encryption::QuickUpdatePageAad(page_aad, page_ordinal_);
-    decryptor->UpdateAad(page_aad);
+    encryption::QuickUpdatePageAad(page_ordinal_, page_aad);
+    decryptor->UpdateAad(*page_aad);
```

Comment:
Was all `const std::string&` changed to `std::string*` because of aad? @pitrou

Reply:
Since I didn't find it neccessary to change to mutable pointer, why do you did this change? Would `page_aad` be nullptr?

Reply:
@pitrou

Reply:
See https://github.com/apache/arrow/pull/15182/commits/8089b4e641d96b3dbceae512d15535efd4f68a31 : the `const std::string&` was actually mutated by the AAD functions!